Hamburger menu

bhushan354 opened the pull request:
- Added proper javascript.
- Added HTML-CSS-JS linters.
- Added proper styling.
- Also used transition while interacting with menu button.

Thank You ,
Bhushan.

V-Blaze commented on app.js:
```diff
@@ -0,0 +1,16 @@
+const menuPageConst = document.querySelector('.menuPage');
+// const pageLinkConst = document.querySelectorAll('.pageLink');
+
+function open() {
+  menuPageConst.style.display = 'flex';
+  menuPageConst.style.top = '0';
+}
+
+function close() {
```
- [ ] You have done great so far, but there are some requirements you still have to implement before moving to the next milestone :+1: :+1: 

- [ ] As stated in the project requirement, you need to try to replicate the Figma design as much as you can, Currently, your Hamburger icon position doesn't match the one in the Figma design, Looking at the Figma you can see that there is a space at the right of the icon but that's not the case In your project. I'll suggest you update this, 
- [ ] Also there seems to be a white space at the right of your mobile design, I'll suggest you look into this too. :+1: :smile: 

- - Your current design:
![Selection_077](https://github.com/bhushan354/Portfolio_Project/assets/107632780/29553314-511d-4b16-a4a8-bad185affbcb)

- - Figma design:
![Screenshot from 2023-06-26 15-05-02](https://github.com/bhushan354/Portfolio_Project/assets/107632780/9a1332c2-76f5-437a-9a24-3cbe639536ba)

bhushan354 replied:
Hello @V-Blaze ,
I have been trying to solve this issue for a long time. can you please tell me with more context how do I solve this issue.

thank you.

V-Blaze commented on app.js:
```diff
@@ -0,0 +1,16 @@
+function close() {
+  menuPageConst.style.top = '-100%';
+}
+
+hamburgerOpenConst.addEventListener('click', open);
+hamburgerCloseConst.addEventListener('click', close);
```
> When the user clicks (or taps) any of the mobile menu options, the mobile menu disappears.
When the user clicks (or taps) any of the mobile menu options, a correct part of the page is displayed.

- [ ] These two requriemtn were not implemented in your project, when a user clicks on a menu item, it should take the user to that particular section of the page. Also when a user clicks on a menu item on a mobile meant, it should also close the menu. I'll suggest you implement these. :smile:

V-Blaze commented on app.js:
```diff
@@ -0,0 +1,16 @@
+const hamburgerOpenConst = document.querySelector('.hamburgerOpen');
+const hamburgerCloseConst = document.querySelector('.hamburgerClose');
+const menuPageConst = document.querySelector('.menuPage');
+// const pageLinkConst = document.querySelectorAll('.pageLink');
+
```
- [ ] following the best practices, you are not meant to leave unused code as a comment in your code base, it makes it look unprofessional, I'll suggest you remove this. :wave:

V-Blaze commented on index.html:
```diff
@@ -4,33 +4,52 @@
       <a href="#" class="myLogo">
-        <img src="MyLogo.png" />
+        <img src="MyLogo.png" alt="logo"/>
       </a>
+         <!-- hamburger section -->
       <nav class="menuButtonMain">
-        <a href="#">
+        <a href="#" class="hamburgerOpen">
```
- [ ] IT seems you have a white space at the right of your mobile screen, this is not in the design and makes your portfolio look a bit unprofessional, I'll suggest you adjust your CSS to fix this as it also affects your mobile menu. :+1: :100: 

- - SCREENSHOT:
![Screenshot from 2023-06-26 15-09-25](https://github.com/bhushan354/Portfolio_Project/assets/107632780/f709d78f-a3ce-45f0-90bb-ccd972afa02f)


- [OPTIONAL] As you build your portfolio gradually, your portfolio is meant to be responsive and look good across different screens, Currently this section in the screenshot below tends to stack on top of each other on some screens, I'll suggest you look into this. :smile: :+1: 

- - SCREENSHOT:
![Screenshot from 2023-06-26 14-57-53](https://github.com/bhushan354/Portfolio_Project/assets/107632780/ebb10919-d420-4e8d-8755-7077953bcbc4)

V-Blaze reviewed (changes requested):
## Hi @bhushan354 , 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

![almost-there](https://user-images.githubusercontent.com/107632780/229242540-644ade4a-fe15-4f4c-a38b-6ded70cc4e8b.gif)


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Correct GitHub flow✔ 
- Meaningful pull request title👋 ✔

 ## Required Changes ♻️

## Optional suggestions

- [OPTIONAL] Currently, it seems you have all your images on the root of your project, this makes it a bit hard to see the main files like `index.html`, I'll suggest creating a new folder like `assets or images` and then move the images into the folder. it will make your project look more organized. :smile: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

bhushan354 commented:
Thank you sir for reviewing my code , I have made changes accordingly.

1) Fixed issue  :=> I have fixed white space in my mobile menu
![image](https://github.com/bhushan354/Portfolio_Project/assets/127464886/3af8038f-cc97-432a-9084-66dc4f3f651d)

2) Fixed issue :=> When a user clicks on a menu item, it should take the user to that particular section of the page.

3) Fixed Issue :=> Removed unused code.

Thank You ,
Bhushan .

usorfaitheloho reviewed (approved):
Hi @bhushan354  👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

![](https://media.giphy.com/media/l49JNZ87F3AmPoH0A/giphy.gif)

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
